lib/resource-mapper.js

```diff
@@ -91,22 +91,25 @@ class ResourceMapper {
     if (filePath.indexOf('/..') >= 0) {
       throw new Error('Disallowed /.. segment in URL')
     }
-    let isIndex = searchIndex && filePath.endsWith('/')
+    let isFolder = filePath.endsWith('/')
 
-    // Create the path for a new file
+    // Create the path for a new ressource
     let path
     if (createIfNotExists) {
       path = filePath
-      // Append index filename if needed
-      if (isIndex) {
-        if (contentType !== this._indexContentType) {
-          throw new Error(`Index file needs to have ${this._indexContentType} as content type`)
+      // create a new file
+      if (!(!searchIndex && isFolder)) {
+        // Append index filename if needed
+        if (searchIndex && isFolder) {
```

Comment:
The sequence of those two `if` conditions is weird; we can do better

Reply:
We already had a discussion about that in PR #1280 that begun this PR #1282 
I initially proposed : 

>       // create a new container
>      if (!searchIndex && isFolder) {
>  	  // else create a new resource
>      } else {

for me it was clearer but you tell me you did not like void if statement.
It was for me the clearest way. The logic being explicit.
Would you prefer this : 

>       // create a new container
>      if (!searchIndex && isFolder) {
>        return { path }             // (do I need the {} around path)
>  	  // else create a new resource
>      } else {

Reply:
Yeah I might have not expressed myself clearly.

What I mean is that

```
if (!(!searchIndex && isFolder)) {
  if (searchIndex && isFolder) {
```

is equivalent to

```
if (searchIndex !! !isFolder) {
  if (isFolder) {
```

Reply:
An other proposal will be : (symbolic code)
> // create a fileIndex
> if (searchIndex && isFolder)  -- > path += this.fileIndex    and check contentType or throw error
> // if create a resource check extension
> if (!isFolder) --> check extension against contentType as actually

as before (create a folder will be implicit), then
> return { path, contentType : contentType || this._defaultContentType }

Reply:
Also fine with me. Just not the repeated conditions.